### PR TITLE
Update installation docs for Python assess beta

### PR DESCRIPTION
**Pull request by drdavella**
This PR contains updates to reflect the release of the beta Python Assess feature. It includes:

* wording to acknowledge the beta release of assess
* updates to installation instructions for assess
* updates to system requirements to reflect C extension build
* a troubleshooting section for installation/runtime issues
* improvements to the middleware inclusion sections to help target non-developers
* fix out-of-date wording about running the service
* updates/improvements to configuration of service and logging

It might be easier to pull these changes and read them in a markdown editor rather than try to parse the diffs.

cc @danilito19 @jleo32 @peter-lazorchak

### CODE REVIEW

**Comment by drdavella** [comment]
Actually, it turns out that github is amazing and you can read the diff *in markdown*. zomg.

File: content/installation/python/PythonConfig.md [hunk 1]
```diff
@@ -137,6 +137,12 @@ Use the properties in this section to override Protect features.
     * **xxe**:
       * **mode**: Set the mode of the rule. Value options are `monitor`, `block`, `block_at_perimeter`, or `off`. <br> Note: If a setting says, "if blocking is enabled", the setting can be `block` or `block_at_perimeter`.
 
+### Contrast Protect properties
```
Comment (drdavella) [inline content/installation/python/PythonConfig.md]: Oops, this should be `Contrast Assess Properties`

File: content/installation/python/PythonConfig.md [hunk 2]
```diff
@@ -90,17 +90,17 @@ Define the following properties to set Syslog values. If the properties aren't d
...
-  * **host**: Set the the hostname or IP address of the Contrast service to which the Contrast agent should report. **Required.** <br> Example: `localhost`
-  * **port**: Set the the port of the Contrast service to which the Contrast agent should report. **Required.** <br> Example: `30555`
+  * **enable**: Set to `false` to prevent the agent from automatically starting the service. This is necessary when connecting to a standalone instance of the service. Defaults to `true`.
+  * **socket**: If this property is defined, and `enable` is set to `true`, the service will be started on a Unix socket at the defined path. Not compatible with the use of `host` and/or `port`. <br> Example: /tmp/service.sock
+  * **host**: If `enable` is `true`, bind the service to this hostname. Otherwise, connect to an external service at this hostname. Defaults to `localhost`. Not compatible with `socket`. <br> Example: `localhost`
+  * **port**: If `enable` is `true`, bind the service to this port. Otherwise, connect to an external service on this port. Defaults to `30555`. Not compatible with `socket`. <br> Example: `30555`
```
Comment (kaitlynmortimer) [inline content/installation/python/PythonConfig.md]: @drdavella What was the original PR for these changes to property descriptions?

File: content/installation/python/PythonInstall.md [hunk 3]
```diff
@@ -80,37 +134,71 @@ if __name__ == '__main__':
...
 app = ContrastMiddleware(app)
 ```
 
 ## Next Steps
 
-Once the installation process is complete, you can update the agent's [configuration](installation-pythonconfig.html) file. 
+Once the installation process is complete, you can update the agent's [configuration](installation-pythonconfig.html) file.
+
+## Troubleshooting
+
```
Comment (kaitlynmortimer) [inline content/installation/python/PythonInstall.md]: Troubleshooting has been permanently moved to the Knowledge Base. I'll remove this section and send it to the CS team.

File: content/installation/python/PythonOverview.md [hunk 4]
```diff
@@ -4,16 +4,24 @@ description: "Troubleshooting the Python Agent"
 tags: "installation python agent troubleshooting"
 -->
 
-The Contrast Python agent provides runtime protection of Django, Flask and Pyramid web applications. 
+The Contrast Python agent provides runtime protection of Django, Flask and Pyramid web applications.
+
+Version 2.3.0 of the agent introduces beta Assess functionality. This provides the ability to find vulnerabilities in applications before they can be exploited.
+
+>**Note:** The beta Assess feature can only be licensed in specific instances of the Contrast UI.
```
Comment (kaitlynmortimer) [inline content/installation/python/PythonOverview.md]: @drdavella What does "specific instances" mean here?

Comment (drdavella) [reply content/installation/python/PythonOverview.md]: The beta version only works in `alpha` and `eval` right now, so it won't be available for all users. There's probably a better way to say that, though.

File: content/installation/python/PythonInstall.md (hunk 3 above)
Comment (drdavella) [reply content/installation/python/PythonInstall.md]: Thank you!

File: content/installation/python/PythonConfig.md (hunk 2 above)
Comment (drdavella) [reply content/installation/python/PythonConfig.md]: It looks like it was #508.

**Pull request merged**